### PR TITLE
kokkos: add HSW, KNL, BDW and Pascal60 support

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

File: var/spack/repos/builtin/packages/kokkos/package.py [hunk 1]
```diff
@@ -62,12 +66,26 @@ def install(self, spec, prefix):
...
                 '--with-serial'
             ]
+            arch_args = []
             if '+openmp' in spec:
                 g_args.append('--with-openmp')
             if 'qthreads' in spec:
                 g_args.append('--with-qthreads=%s' % spec['qthreads'].prefix)
             if 'cuda' in spec:
                 g_args.append('--with-cuda=%s' % spec['cuda'].prefix)
+            # 1)  hopefully we have a host architecture
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/kokkos/package.py]: I don't really like "hopefully"s. We should check this is the case with a `conflicts` directive.

Can `--arch` accept multiple architectures? We should use a multi-valued variant here.

Comment (junghans) [reply var/spack/repos/builtin/packages/kokkos/package.py]: You are right, to enable `Pascal60` one of them need to be on.

Comment (womeld) [reply var/spack/repos/builtin/packages/kokkos/package.py]: Thank you @adamjstewart and @junghans. A suggested template was Charm which I will reference.

Comment (citibeth) [reply var/spack/repos/builtin/packages/kokkos/package.py]: Even if not a multi-valued variant, it should be a string-type variant.

**Comment by junghans** [comment]
Superseded by #8226.

**Pull request closed without merging**